iahrs_driver:     # node name
  ros__parameters:
    tf:
      frame_id: iahrs_link
      parent_frame_id: base_link

    serial_port: "/dev/sensors/iahrs"
    serial_baud: B115200

    is_debug: false
    publish_imu: true
    publish_mag: false
    publish_temp: false

    object:
      info: vr    # product and version info.
      help: h     # display help; hh - display help for filters
      volt: pv    # power in voltage[V]
      temp: t     # temperature['C]
      racc: a     # acceleration[g] in sensor coordinate frame
      rgyr: g     # gyroscope[deg/s] in sensor coordinate frame
      rmag: m     # magnetometer[uT] in sensor coordinate frame
      eule: e     # Euler angles(Z->Y->X rotation)
      quat: q     # Quaternion(r, v0, v1, v2)
      gacc: ag    # acceleration-gravity(ax, ay, az) [m/s2]
      gvel: v     # velocity(vx, vy, vz) [m/s]
      gpos: p     # position(px, py, pz) [m]
      frqm: f     # freqency and magnitube(fx, mx, fy, my, fz, mz)
      vibd: vd    # vibration frequency and damping ratio, magnitude
      stat: s     # status
      cont: c     # control command
      optn: o     # system options
      baud1: b1   # RS-232/UART1 baudrate[bps] *
      baud2: b2   # USB/UART2 baudrate[bps] *
      sprt: so    # sync. data transfer port (0 - RS-232/UART1, 1 - USB/UART2)
      sprd: sp    # sync. data transfer period (1 ~ 60000) [ms]
      sdat: sd    # sync. data selection (0 ~ 0xFFF)
      sync_data_type:
        stmp: 0x0001    # timecount [ms]
        temp: 0x0002    # temperature['C]
        racc: 0x0004    # acceleration[g]
        rgyr: 0x0008    # gyroscope[deg/s]
        rmag: 0x0010    # magnetometer[uT]
        gacc: 0x0020    # acceleration-gravity(ax, ay, az) [m/s2]
        eule: 0x0040    # Euler angles(Z->Y->X rotation)
        quat: 0x0080    # Quaternion(r, v0, v1, v2)
        gvel: 0x0100    # velocity(vx, vy, vz) [m/s]
        gpos: 0x0200    # position(px, py, pz) [m]
        frqm: 0x0400    # freqency and magnitube(fx, mx, fy, my, fz, mz)

      fftsrc: fs        # FFT source(0: none, 1: Accl. 2: Gyro.)
      vibsrc: vs        # Vibration source(0: none, 1: Accl. 2: Gyro.)
      control:
        savconf: fw     # (c=1) save configuration parameters to flash memory
        rldfact: fd     # (c=2) reload factory default values
        caliacc: ca     # (c=3) calibrate bias&scale of accelerometer&gyroscope
        calimag: cm     # (c=4) calibrate bias of magnetometer
        zeroeul: za     # (c=5) set current Euler angle data to zero
        rseteul: ra     # (c=7) reset Euler angle
        rsetpove: rp    # (c=8) reset position & velocity
        rsetcal: rc     # (c=9) reset all calibration data
        restart: rd     # (c=99) restart device

    ### Commands marked with * are applied after restart.
    compensater:
      gyro:
        full_scale: gs    # gyroscope full scale (0-250dps, 1-500dps, 2-1000dps, 3-2000dps) *
        lp_filter:  gl    # gyroscope low-pass filter (-1- bypass, 0 ~ 7 - LPF cutoff freq.) *
        gdrf_thres: dt    # gyrosocope drift reduction filter threshold (0 ~ 360)
        gdrf_gain: dg     # gyrosocope drift reduction filter gain (0 ~ 360)
      accelro:
        full_scale: as    # accelerometer full scale (0-2g, 1-4g, 2-8g, 3-16g) *
        lp_filter:  al    # accelerometer low-pass filter (-1- bypass, 0 ~ 7 - LPF cutoff freq.) *
        asaf_thres: st    # acceleration scale adjustment filter threshold (0 ~ 1)
        asaf_gain: sg     # acceleration scale adjustment filter gain (0 ~ 1)
      kalman:
        accelero: av      # Kalman filter variance for accelerometer (0.001 ~ 1000; 0-disable)
        magneto: mv       # Kalman filter variance for magnetometer (0.001 ~ 1000; 0-disable)
        zero_velo: zv     # Kalman filter variance for zero velocity (0.001 ~ 1000; 0-disable)

    # linear_acceleration_covariance, angular_velocity_covariance, orientation_covariance
    imu_data:
      lin_acc_cov: [0.0064, 0.0, 0.0,
                    0.0, 0.0063, 0.0,
                    0.0, 0.0, 0.0064]
      ang_vel_cov: [0.000559, 0.0, 0.0,
                    0.0, 0.000489, 0.0,
                    0.0, 0.0, 0.000105]
      ang_vel_cov_deg: [0.032, 0.0, 0.0,
                    0.0, 0.028, 0.0,
                    0.0, 0.0, 0.006]
      orient_cov: [0.000227, 0.0, 0.0,
                    0.0, 0.000192, 0.0,
                    0.0, 0.0, 0.000105]
      orient_cov_deg: [0.013, 0.0, 0.0,
                    0.0, 0.011, 0.0,
                    0.0, 0.0, 0.006]

    spec:
      voltage:        # v
        low: 4.2
        high: 12
      temp:           # 'c
        low: 0
        high: 45
        xlow: -40
        xhigh: 85
      range:
        gyro: 2000       # degree per second
        accelero: 16     # g
        magneto: 4900    # uT
      resolution:
        static_roll: 0.012     # deg/h
        static_pitch: 0.012    # deg/h
        static_yaw: 8.6        # deg/h
        dynamic_roll: 0.033    # deg/h
        dynamic_pitch: 0.183   # deg/h
        dynamic_yaw: 10.264    # deg/h




